### PR TITLE
Merge upstream 5.6.37 into Percona Server

### PR DESCRIPTION
**Pull request by percona-ysorokin**

### CODE REVIEW

**Comment by percona-ysorokin** [comment]
Jenkins build link
http://jenkins.percona.com/job/percona-server-5.6-param/1967/
(all release nodes are marked red for some reason although MTR runs succeeded)

File: scripts/CMakeLists.txt [hunk 1]
```diff
@@ -124,6 +124,12 @@ STRING(REPLACE       "-fstack-protector-strong"      "" CXXFLAGS "${CXXFLAGS}")
 STRING(REPLACE       "-grecord-gcc-switches"         "" CXXFLAGS "${CXXFLAGS}")
 STRING(REPLACE       "-pipe"                         "" CXXFLAGS "${CXXFLAGS}")
 
+IF(CMAKE_SYSTEM_NAME MATCHES "FreeBSD")
+  SET (PERL_PATH "/usr/local/bin/perl")
```
Comment (laurynas-biveinis) [inline scripts/CMakeLists.txt:128]: Our scripts/mysqldumpslow.sh also falls under the the s/#!/usr/bin/perl/@PERL_PATH@ /g replacement (I have no idea why upstream only changed scripts dir when there are Perl scripts everywhere)

**Review by laurynas-biveinis — CHANGES_REQUESTED** [review]
Automatic LGTM with scripts/mysqldumpslow.sh converted to @PERL_PATH@. No need for full test cycle: cmake + inspection of @PERL_PATH@ replacement in the script is enough

File: scripts/CMakeLists.txt (hunk 1 above)
Comment (percona-ysorokin) [reply scripts/CMakeLists.txt:128]: fixed

**Pull request merged**